src/main/java/com/rollbar/android/Notifier.java

```diff
@@ -279,25 +281,35 @@ public void onFailure(HttpResponse response) {
                 Log.e(Rollbar.TAG, "There was a problem reporting to Rollbar.");
                 Log.e(Rollbar.TAG, "Response: " + response);
 
-                if (file == null) {
-                    if (!response.hasStatusCode()) {
+                // if there's no response code or the response code is not a 2xx or 4xx
+                //    write a file if one doesn't exist already and re-schedule
+                // TODO(eric): make this conditional check one function
+                if (!response.hasStatusCode() || !(response.has2XXStatusCode() || response.has4XXStatusCode())) {
+                    Log.d(Rollbar.TAG, "Failed response: write new file or use existing");
+                    if (file == null) {
+                        Log.d(Rollbar.TAG, "Failed response: writing a new file.");
                         writeItems(items);
                     }
+
+                    Log.d(Rollbar.TAG, "Failed response: re-scheduling.");
+                    scheduleItemFileHandler(itemScheduleDelay);
+                // otherwise, delete the file if it exists
                 } else {
-                    // Give up if there is a server error
-                    if (response.hasStatusCode()) {
+                    Log.d(Rollbar.TAG, "Failed response: don't write new file and delete if it exists");
```

Comment:
This confused me until I read the code a bunch of times. Am I correct that the intention is:

- if we get a 5xx response from Rollbar, assume that it is not going to succeed on retry. so abort instead of retrying again
- since we are aborting, delete the file to clean it up.

Assuming this is correct, I suggest cleaning up these debug messages to make it more clear. It's confusing to me that this case and the one above both say "Failed response" even though they are handling 2 very different cases.

Also, is it correct that `file` only contains the one rollbar payload? (i.e. it's one file per payload, not a file that's used as a queue?)